src/main/groovy/org/boozallen/plugins/jte/init/primitives/TemplatePrimitiveInjector.groovy

```diff
@@ -93,22 +97,117 @@ abstract class TemplatePrimitiveInjector implements ExtensionPoint{
     void validateConfiguration(FlowExecutionOwner flowOwner, PipelineConfigurationObject config){}
 
     /**
-     * parse the aggregated pipeline configuration to instantiate a {@link TemplatePrimitive} and store it in
-     * the {@link TemplateBinding}
+     * parse the aggregated pipeline configuration to instantiate a {@link TemplatePrimitive} and store it
+     * in a TemplatePrimitiveNamespace
      *
      * @param flowOwner the run's flowOwner
      * @param config the aggregated pipeline configuration
-     * @param binding the run's common {@link TemplateBinding}
      */
-    void injectPrimitives(FlowExecutionOwner flowOwner, PipelineConfigurationObject config, TemplateBinding binding){}
+    void injectPrimitives(FlowExecutionOwner flowOwner, PipelineConfigurationObject config){}
 
     /**
-     * A second pass allowing the different injector's to inspect what is in the binding and respond accordingly
+     * A second pass allowing the different injector's to inspect what TemplatePrimitives
+     * have been created and respond accordingly
      *
      * @param flowOwner the run's flowOwner
      * @param config the aggregated pipeline configuration
-     * @param binding the run's common {@link TemplateBinding}
      */
-    void validateBinding(FlowExecutionOwner flowOwner, PipelineConfigurationObject config, TemplateBinding binding){}
+    void validatePrimitives(FlowExecutionOwner flowOwner, PipelineConfigurationObject config){}
+
+    TemplatePrimitiveCollector getPrimitiveCollector(FlowExecutionOwner flowOwner){
+        WorkflowRun run = flowOwner.run()
+        if(!run){
+            throw new JTEException("Invalid Context. Cannot determine run.")
+        }
+        TemplatePrimitiveCollector primitiveCollector = run.getAction(TemplatePrimitiveCollector)
+        if(primitiveCollector == null){
+            primitiveCollector = new TemplatePrimitiveCollector()
```

Comment:
Dont fully understand the action part so, asking, if `run.getAction(TemplatePrimitiveCollector)` is null, do we want to insert this instance as run.getAction(TemplatePrimitiveCollector) or should each call to getPrimitiveCollecter return a new instance